src/MiniProfiler.Shared/ProfileProviders/BaseProfilerProvider.cs

```diff
@@ -29,6 +29,11 @@ public abstract class BaseProfilerProvider : IProfilerProvider
         public abstract MiniProfiler GetCurrentProfiler();
 
         /// <summary>
+        /// GEts or sets the current head timing. This is used by <see cref="MiniProfiler.Head"/>.
+        /// </summary>
+        public abstract Timing CurrentHead { get; set; }
```

Comment:
I understand your concern, my thinking was that there could be more than one head timing in an async scenario. However in the current design whether async scenario is supported is left up to the profiler provider, so it looks like the only way to support is to store current head in the profiler provider. Do you think it would be better to move the AsyncLocal backing the head into mini profiler?